### DISCUSSION TITLE
How many services should be there?

### DISCUSSION THREAD

**Discussion by kaustubhn** [P0W/option_strategies#10] [Q&A]
@P0W This question comes because we need to decide how many containers need to be run, depending on that we will keep the containers, if you can list down the services and briefly list the goal of each service I can build those containers.

**Comment by kaustubhn** [comment]
Service Suggestions:
1. Live Feed Manager 
2. Client Manager 
3. Order Manager 
4. Strategy
5. Redis -> Queue
6. MongoDB -> Database

**Comment by P0W** [comment]
Those looks good, as uber level goal.

But currently its monolithic in following sense:

1. The "Strategy" would need an instance of Live Feed manager, Strike Manager and Order Manager
2. All manager would need an instance of IClientManager

We can containerize following:
1. Telegram Bot (`src\bot\bot.py`) - which requires an instance of `iclientmanager` to query orders (which technically simply rely on mongodb instance)
2. Mongodb - to write placed order status, telegram bot will use this as well
3. Redis - to store user access token used from login method of clients
4. Dummy feed (`src\simulator\feeds.py`)

This is currently I am thinking on top off my head.

To see the look and feel of entry point refer `tests\strategy_test.py` main

**Reply by kaustubhn** [reply]
This should be good, I am not against monolithic architectures, I just thought that if its monolithic then we should have all the code base in a single package/folder. Since its easy to package & run in container.

I will consider this and make some changes/add containers.
